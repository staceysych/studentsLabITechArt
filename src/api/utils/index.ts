import { postRequest } from "./post-request";

export { postRequest };
