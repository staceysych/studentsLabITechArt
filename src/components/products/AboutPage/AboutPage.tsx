import React from "react";

const AboutPage: React.FC = () => <h3>About</h3>;

export default AboutPage;
