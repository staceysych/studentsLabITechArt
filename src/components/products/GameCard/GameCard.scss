@import "../../../styles/variables.scss";

.GameCard {
  width: 200px;
  height: 390px;
  margin: 10px;
  border-radius: 5px;
  box-shadow: 1px 9px 10px 0 $basicColorBright;
  transition: 0.3s all ease-out;

  &__rating {
    width: 100%;
    height: 30px;
    display: flex;
    justify-content: flex-start;

    img {
      margin-right: 3px;
      height: 100%;
      width: 15%;
    }
  }

  &__img {
    width: 100%;
    height: 60%;
    overflow: hidden;

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
      border-top-left-radius: 5px;
      border-top-right-radius: 5px;
      transition: 0.3s all ease-out;
      &:hover {
        transform: scale(1.2);
        cursor: pointer;
      }
    }
  }

  h2 {
    font-size: 16px;
    text-transform: capitalize;
    margin-top: 0;
  }

  span {
    display: block;
    margin-top: 10px;
    font-weight: bold;
    font-size: 14px;
    position: absolute;
    bottom: 0;
    width: 100%;
    left: 0;
    padding: 5px 0;
    border-bottom-right-radius: 5px;
    border-bottom-left-radius: 5px;
    background-color: $highlightColor;
  }

  &__content {
    height: 40%;
    padding: 10px;
    text-align: center;
    position: relative;
  }

  &__controls {
    display: flex;
    justify-content: space-around;
    align-items: center;
    margin-top: 5px;
    &_edit {
      width: 70px;
      height: 30px;
      background: $updateColor;
      border-radius: 5px;
      outline: none;
      cursor: pointer;
      font-size: 18px;
      padding: 5px;
      display: flex;
      align-items: center;
      justify-content: center;
      color: $textColor;
      border: none;
      transition: 0.3s all ease-in-out;
      &:hover {
        background: none;
        border: 2px solid $updateColor;
        color: $updateColor;
      }
    }
    &_delete {
      width: 70px;
      height: 30px;
      background: none;
      border: 2px solid $errorColorBG;
      border-radius: 5px;
      outline: none;
      cursor: pointer;
      font-size: 18px;
      padding: 5px;
      display: flex;
      align-items: center;
      justify-content: center;
      color: $errorColorBG;
      transition: 0.3s all ease-in-out;

      &:hover {
        background: $errorColorBG;
        color: $textColor;
      }
    }
  }
}
