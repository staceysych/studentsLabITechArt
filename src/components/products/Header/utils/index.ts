import { dropDownData } from "./drop-down-data";

export { dropDownData };
