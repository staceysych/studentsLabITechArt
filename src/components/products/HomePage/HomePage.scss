.HomePage {
  &__content {
    padding-top: 30px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    padding-bottom: 140px;
  }
}
