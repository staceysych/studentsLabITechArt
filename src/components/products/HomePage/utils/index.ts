import { searchGame } from "./searchGame";
import { getTopProducts } from "./getTopProducts";
import { generateTitle } from "./generateTitle";

export { searchGame, getTopProducts, generateTitle };
