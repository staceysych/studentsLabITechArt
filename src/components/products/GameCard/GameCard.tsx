import React from "react";
import { useDispatch, useSelector } from "react-redux";

import "./GameCard.scss";

import star from "images/star.svg";

import { IGameObject, RootState } from "../../../utils/interfaces";

import { PAGE_ACTIONS } from "../../../redux/actions/creators";

import { URLS } from "../../../constants";

interface Props {
  obj: IGameObject;
}

const GameCard: React.FC<Props> = ({ obj: { id, poster, name, rating, price } }) => {
  const dispatch = useDispatch();
  const products = useSelector((state: RootState) => state.page.products);

  const addToCart = () => {
    dispatch(PAGE_ACTIONS.getCartProducts(`${URLS.SERVER_URL}${URLS.GET_PRODUCT_BY_ID_URL}${id}`));
  };

  return (
    <div className="GameCard" key={id} onClick={() => addToCart()} aria-hidden="true">
      <div className="GameCard__img">
        <img src={poster} alt={name} />
      </div>
      <div className="GameCard__content">
        <h2>{name}</h2>
        <div className="GameCard__rating">
          {[...Array(rating)].map(() => (
            <img key={Math.random()} alt={rating.toString()} src={star} />
          ))}
        </div>
        <span>{`${price} BYN`}</span>
      </div>
    </div>
  );
};

export default GameCard;
