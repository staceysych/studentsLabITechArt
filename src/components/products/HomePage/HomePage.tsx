import React, { useState, useEffect } from "react";

import SearchBar from "../SearchBar/index";
import GameCard from "../GameCard/index";

import "./HomePage.scss";

import { searchGame, getTopProducts, generateTitle } from "./utils/index";
import { useDebounce, IGameObject } from "../../../utils/index";
import { CONSTANTS } from "../../../constants/index";

const HomePage: React.FC = () => {
  const { DEBOUNCE_TIME } = CONSTANTS;
  const [searchText, setSearchText] = useState<string>("");
  const [results, setResults] = useState<Array<object>>([]);
  const [isSearching, setIsSearching] = useState<boolean>(false);
  const [recentGames, setRecentGames] = useState<Array<object>>([]);

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setIsSearching(true);
    setSearchText(event.target.value);
  };

  const debouncedSearchText = useDebounce(searchText, DEBOUNCE_TIME);

  useEffect(() => {
    getTopProducts().then((data) => {
      setRecentGames(data);
    });
  }, []);

  useEffect(() => {
    if (debouncedSearchText) {
      setIsSearching(true);
      searchGame(debouncedSearchText).then((data) => {
        setIsSearching(false);
        setResults(data);
      });
    } else {
      setResults([]);
      setIsSearching(false);
    }
  }, [debouncedSearchText]);

  return (
    <>
      <SearchBar {...{ handleChange, isSearching }} />
      {!isSearching && <h2 className="HomePage__title">{generateTitle(searchText, results)}</h2>}
      {isSearching && <h2 className="HomePage__title">Searching for results...</h2>}
      <div className="HomePage__content">
        {results.length && !isSearching
          ? results.map((obj: IGameObject) => <GameCard key={Math.random()} obj={obj} />)
          : null}
        {!searchText && !results.length
          ? recentGames.slice(-3).map((obj: IGameObject) => <GameCard key={Math.random()} obj={obj} />)
          : null}
      </div>
    </>
  );
};

export default HomePage;
