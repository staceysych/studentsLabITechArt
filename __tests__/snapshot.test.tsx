import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import AboutPage from "../src/components/products/AboutPage";
import HomePage from "../src/components/products/HomePage";
import ProductsPage from "../src/components/products/ProductsPage";
import GameCard from "../src/components/products/GameCard";

import { createMockStore } from "../src/helpers/testHelpers";

import { getStateForTesting } from "../src/utils";

const mockStore = createMockStore();

describe("Snapshot testing", () => {
  let store;
  let component;

  beforeEach(() => {
    const state = getStateForTesting();

    store = mockStore(state);

    store.dispatch = jest.fn();
  });

  it("AboutPage renders correctly", () => {
    component = renderer.create(
      <Provider store={store}>
        <AboutPage />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("HomePage renders correctly", () => {
    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]}>
          <HomePage />
        </MemoryRouter>
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("ProductsPage renders correctly", () => {
    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]}>
          <ProductsPage />
        </MemoryRouter>
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("GameCard renders correctly", () => {
    const mockedProps = {
      id: 1,
      name: "GTA",
      rating: 5,
      price: 120,
      poster: "yandex.ru",
      date: "2010-08-12",
      genre: "shooter",
      age: "18",
      devise: "Xbox",
    };
    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]}>
          <GameCard obj={mockedProps} />
        </MemoryRouter>
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
