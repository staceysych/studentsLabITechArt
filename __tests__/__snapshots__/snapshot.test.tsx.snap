// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing AboutPage renders correctly 1`] = `
<h3>
  About
</h3>
`;

exports[`Snapshot testing GameCard renders correctly 1`] = `
<div
  className="GameCard"
>
  <div
    aria-hidden="true"
    className="img"
    onClick={[Function]}
  >
    <img
      alt="GTA"
      src="yandex.ru"
    />
  </div>
  <div
    className="content"
  >
    <h2>
      GTA
    </h2>
    <div
      className="rating"
    >
      <img
        alt="5"
        src={Object {}}
      />
      <img
        alt="5"
        src={Object {}}
      />
      <img
        alt="5"
        src={Object {}}
      />
      <img
        alt="5"
        src={Object {}}
      />
      <img
        alt="5"
        src={Object {}}
      />
    </div>
    <span>
      120 BYN
    </span>
  </div>
</div>
`;

exports[`Snapshot testing HomePage renders correctly 1`] = `
<div
  className="HomePage"
>
  <div
    className="controls"
  >
    <div
      className="wrapper"
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
      <button
        className="btn"
        type="button"
      >
        <img
          alt="lens"
          src={Object {}}
        />
      </button>
    </div>
  </div>
  <h2
    className="title"
  >
    The most recent games added to the website
  </h2>
  <div
    className="content"
  >
    <div
      className="GameCard"
    >
      <div
        aria-hidden="true"
        className="img"
        onClick={[Function]}
      >
        <img
          alt=""
          src=""
        />
      </div>
      <div
        className="content"
      >
        <h2>
          
        </h2>
        <div
          className="rating"
        />
        <span>
          0 BYN
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot testing ProductsPage renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="header"
  >
    <h1 />
    <div
      className="controls"
    >
      <div
        className="wrapper"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="Search"
          type="text"
        />
        <button
          className="btn"
          type="button"
        >
          <img
            alt="lens"
            src={Object {}}
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="sortContainer"
    >
      <div
        className="sort"
      >
        <h2>
          Sort:
        </h2>
        <div
          className="sortField"
        >
          <h4>
            Criteria: 
          </h4>
          <div
            className="wrapper"
          >
            <div
              className="text"
              onClick={[Function]}
            >
              Select
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <h4>
            Type: 
          </h4>
          <div
            className="wrapper"
          >
            <div
              className="text"
              onClick={[Function]}
            >
              Select
            </div>
          </div>
        </div>
        <h3>
          Genre: 
        </h3>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle false"
            >
              <div
                className="innerCircle false"
              />
            </div>
            <div
              className="label"
            >
              all
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              shooter
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              fighting
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              racing
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              strategy
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              sport
            </div>
          </div>
        </div>
        <h3>
          Age: 
        </h3>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle false"
            >
              <div
                className="innerCircle false"
              />
            </div>
            <div
              className="label"
            >
              all
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              6
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              12
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              16
            </div>
          </div>
        </div>
        <div
          className="sortField"
        >
          <div
            className="wrapper"
            onClick={[Function]}
          >
            <div
              className="outerCircle unselected"
            >
              <div
                className="innerCircle unselectedCircle"
              />
            </div>
            <div
              className="label"
            >
              18
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="productsContainer"
    >
      <div
        className="products"
      >
        <div
          className="wrapper"
        >
          <div
            className="bounce1"
          />
          <div
            className="bounce2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
